### PR TITLE
ddl: fix auto_increment is not compatible with MySQL

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve? 
#13032

### What is changed and how it works?
The isKet and auto_increment checks must be paired.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

None

Side effects

None

Related changes

None

Release note

None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

**Comment by bb7133** [comment]
hi @Rustin-Liu Thank you for your contribution!

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -315,32 +312,36 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
...
-		isKey = isConstraintKeyTp(stmt.Constraints, autoIncrementCol)
-	}
-	autoIncrementMustBeKey := true
-	for _, opt := range stmt.Options {
-		if opt.Tp == ast.TableOptionEngine && strings.EqualFold(opt.StrValue, "MyISAM") {
-			autoIncrementMustBeKey = false
-		}
-	}
-	if (autoIncrementMustBeKey && !isKey) || count > 1 {
+	} else if len(autoIncrementCols) > 1 {
```
Comment (zimulala) [inline planner/core/preprocess.go]:
```suggestion
	if len(autoIncrementCols) > 1 {
```
The `if` statement is returned in line320.

File: planner/core/preprocess.go [hunk 2]
```diff
@@ -315,32 +312,36 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
...
 		p.err = autoid.ErrWrongAutoKey.GenWithStackByArgs()
```
Comment (zimulala) [inline planner/core/preprocess.go]: Here we also can add `return`. Then we can remove the `else` statement on line323 to reduce one level of code nesting.

File: planner/core/preprocess.go [hunk 3]
```diff
@@ -315,32 +312,36 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
...
-	}
-
-	switch autoIncrementCol.Tp.Tp {
-	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeLong,
-		mysql.TypeFloat, mysql.TypeDouble, mysql.TypeLonglong, mysql.TypeInt24:
-	default:
-		p.err = errors.Errorf("Incorrect column specifier for column '%s'", autoIncrementCol.Name.Name.O)
+	} else {
+		// Only have one auto_increment col.
+		for col, isKey := range autoIncrementCols {
```
Comment (zimulala) [inline planner/core/preprocess.go]: If it only one auto_increment col, can we remove this `for`?

File: planner/core/preprocess.go (hunk 1 above)
Comment (Rustin-Liu) [reply planner/core/preprocess.go]: refactored.

File: planner/core/preprocess.go (hunk 2 above)
Comment (Rustin-Liu) [reply planner/core/preprocess.go]: refactored.

File: planner/core/preprocess.go (hunk 3 above)
Comment (Rustin-Liu) [reply planner/core/preprocess.go]: I think we can keep it, because it's simple and clear enough.

**Comment by zimulala** [comment]
PTAL @Deardrops @wjhuang2016

**Pull request merged**